### ISSUE TITLE
Adding items to the grid fails

### ISSUE THREAD

**Issue by tquetano-r7** [STRML/react-grid-layout#368]
When adding items to the grid I receive this error:

```javascript
utils.js:50 Uncaught (in promise) TypeError: Cannot read property 'length' of undefined
```

In the `synchronizeLayoutWithChildren` method, in the bottom catchall scenario it is sending `layout` to the `bottom` method, however `layout` is not yet defined (as it is being assigned).  It appears that [this commit](https://github.com/STRML/react-grid-layout/commit/eac7e0817ab71813c413b1e22f387ccf66ec64ce) is the source, because layout was an instantiated array prior (and was therefore sending the `layout` in-progress with each execution), and is now leveraging `React.Children.map`.

**Comment by STRML** [comment]
Thanks! This is certainly a bug, let me see what I can do.

**Comment by STRML** [comment]
Published in 0.13.8 - thanks for the detailed report!

**Comment by tquetano-r7** [comment]
16 minutes ... that's some kind of record! Thanks!
